src/main/server/fix_mongodb_for_upgrading_from_2.4.1_and_earlier.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# In version 2.4.2, the MongoDB driver is updated. 
+# If you use MongoDB database and upgrade your server from v2.4.1 or below versions to the latest version,
+# you should run this script
+# Here is the script usage:
+# ./fix_mongodb_for_upgrading_from_2.4.1_and_earlier.sh MONGODB_IP MONGODB_USERNAME MONGODB_PASSWORD
+  
+MONGODB_IP=$1
+MONGODB_USERNAME=$2
+MONGODB_PASSWORD=$3
+TODAY=$(date +"%Y-%m-%d-%H-%M")
+
+
+usage() {
+  #show usage 
+  echo "In version 2.4.2, the MongoDB driver is updated."
+  echo "If you use MongoDB database and upgrade your server "
+  echo "from v2.4.1 or below versions to the latest version, you should run this script"
+  #install mongodb-clients 
+  echo "Usage:"
+  echo "$0 MONGODB_IP MONGODB_USERNAME MONGODB_PASSWORD\n"
+  
+}
+
+
+run_mongo() {
+    db=$1
+    shift
+    mongo "$MONGODB_IP/$db" --quiet --eval "$@"
+}
+
+run_mongo_auth() {
+        db=$1
+        shift
+        mongo --host $MONGODB_IP --authenticationDatabase admin $db --quiet --username $MONGODB_USERNAME --password $MONGODB_PASSWORD  --eval "$@"
+}
+
+mongo_backup(){
+        if [ -z "$MONGODB_USERNAME" ] && [ -z "$MONGODB_PASSWORD" ]; then
+                mongodump --host $MONGODB_IP --gzip -o mongo_dump_$TODAY
+        elif [ ! -z "$MONGODB_USERNAME" ] && [ ! -z "$MONGODB_PASSWORD" ]; then
+                mongodump --host $MONGODB_IP --username $MONGODB_USERNAME --password $MONGODB_PASSWORD --gzip -o mongo_dump_$TODAY
+        fi
+
+        if [ $? != 0 ]; then
+           echo "The backup is not completed so it does not continue."
+           echo "Please check credentials are correct and you've write permission to current directory: `pwd` "
+           echo "If it's a permission issue, you may need to run this command with sudo"
+           exit 1;
+        else
+           echo "The backup is completed. Backup Folder: /usr/local/antmedia/mongo_dump_$TODAY"
+        fi
+}
+
+
+
+
+mongocon() {
+        if [ -z "$MONGODB_USERNAME" ] && [ -z "$MONGODB_PASSWORD" ]; then
```

Comment:
Handling username and password is good. It would be better if there is single command instead of having 'run_mongo' and 'run_mongo_auth'. It decrease the code duplication